contracts/icp/proxy-contract/src/mutate.rs

```diff
@@ -0,0 +1,304 @@
+use std::collections::HashSet;
+
+use candid::{CandidType, Principal};
+
+use crate::types::*;
+use crate::PROXY_CONTRACT;
+
+async fn check_member(_signer_id: &ICSignerId) -> Result<bool, String> {
+    // let context_canister_id = PROXY_CONTRACT.with(|contract| {
+    //     contract.borrow().context_config_id.clone()
+    // });
+
+    // let principal = identity_to_principal(&signer_id.0);
+    // TODO: implement this
+    // let call_result: CallResult<(bool,)> = ic_cdk::call(
+    //     Principal::from_text(&context_canister_id)
+    //         .map_err(|e| format!("Invalid context canister ID: {}", e))?,
+    //     "is_member",
+    //     (principal,),
+    // ).await.map_err(|e| format!("Failed to call context contract: {:?}", e))?;
+
+    Ok(true)
+}
+
+#[ic_cdk::update]
+async fn mutate(
+    signed_request: ICPSigned<ICRequest>,
+) -> Result<Option<ICProposalWithApprovals>, String> {
+    let request = signed_request
+        .parse(|r| r.signer_id)
+        .map_err(|e| format!("Failed to verify signature: {}", e))?;
+
+    // Check request timestamp
+    let current_time = ic_cdk::api::time();
+    if current_time.saturating_sub(request.timestamp_ms) > 1000 * 5 {
+        return Err("request expired".to_string());
+    }
+
+    // Check membership
+    if !check_member(&request.signer_id).await? {
+        return Err("signer is not a member".to_string());
+    }
+
+    match &request.kind {
+        ICRequestKind::Propose { proposal } => {
+            let num_proposals = PROXY_CONTRACT.with(|contract| {
+                let contract = contract.borrow();
+                contract
+                    .num_proposals_pk
+                    .get(&proposal.author_id)
+                    .copied()
+                    .unwrap_or(0)
+            });
+
+            internal_create_proposal(proposal.clone(), num_proposals)
```

Comment:
`internal_create_proposal` already reads `PROXY_CONTRACT.with`, why not fetch it there?

```suggestion
            internal_create_proposal(proposal.clone()
```